syntax = "proto3";

package main;

service MainService {
    rpc ProcessOrder(ProcessOrderRequest) returns (ProcessOrderResponse);
    rpc GetOrderStatus(GetOrderStatusRequest) returns (GetOrderStatusResponse);
}

message ProcessOrderRequest {
    string user_id = 1;
    repeated OrderItem items = 2;
}

message OrderItem {
    string product_id = 1;
    int32 quantity = 2;
    double price = 3;
}

message ProcessOrderResponse {
    string order_id = 1;
    string status = 2;
    string message = 3;
}

message GetOrderStatusRequest {
    string order_id = 1;
}

message GetOrderStatusResponse {
    string order_id = 1;
    string status = 2;
    repeated OrderItem items = 3;
    double total_amount = 4;
    string created_at = 5;
}
